mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala

```diff
@@ -146,11 +151,29 @@ object Keyword {
       .toList match {
       // (class|trait|object) classname ext@@
       case (_: Token.Ident) :: (_: Token.Ident) :: kw :: Nil =>
-        if (kw.is[Token.KwClass] || kw.is[Token.KwTrait] || kw.is[Token.KwObject]) true
+        if (kw.is[Token.KwClass] || kw.is[Token.KwTrait] || kw.is[Token.KwObject] || kw.is[Token.KwEnum]) true
         else false
       // ... classname() ext@@
       case (_: Token.Ident) :: (_: Token.RightParen) :: _ => true
+      // ... classname[T] ext@@
+      case (_: Token.Ident) :: (_: Token.RightBracket) :: _ => true
+      case _ => false
+    }
+  }
+
+  private def derivesAfterExtends(leadingReverseTokens: Array[Token]): Boolean = {
+    leadingReverseTokens
+      .filterNot(t => t.is[Token.Whitespace] || t.is[Token.EOF])
+      .take(4)
+      .toList match {
+      // ... extends ... with A der@@
```

Comment:
We could potentially have a very similar logic for `with`

The only issue I see is that if we have `val a: A with B der@@` which is not valid, but maybe it;s not an issue since this will not be witten this way very often

Reply:
Ah, this case should be deleted, since the one below covers it